Formula/katago.rb

```diff
@@ -0,0 +1,36 @@
+class Katago < Formula
+  desc "Neural Network Go engine with no human-provided knowledge"
+  homepage "https://github.com/lightvector/KataGo"
+  url "https://github.com/lightvector/KataGo/archive/v1.3.2.tar.gz"
+  sha256 "75e783cd2db6180f2cc694703df809e5c04ba24c09546a9c24aad575e18ef2d8"
+
+  depends_on "cmake" => :build
+  depends_on "boost"
+  depends_on "libzip"
+
+  resource "15b-network" do
+    url "https://github.com/lightvector/KataGo/releases/download/v1.3.2/g170e-b15c192-s1672170752-d466197061.txt.gz", :using => :nounzip
+    sha256 "09456899f1b9155217f4ca059cf8a68f79b7eba22cf6b7c00ffb7f17ce067967"
+  end
+
+  resource "20b-network" do
+    url "https://github.com/lightvector/KataGo/releases/download/v1.3.2/g170-b20c256x2-s1913382912-d435450331.txt.gz", :using => :nounzip
+    sha256 "c309c0361ca1109084dd026235c9fa0254ffe22eecf096b6c5988a9d902a75ca"
+  end
+
+  def install
+    cd "cpp" do
+      system "cmake", ".", "-DBUILD_MCTS=1", "-DUSE_BACKEND=OPENCL", "-DNO_GIT_REVISION=1", "-DCMAKE_OSX_SYSROOT=#{MacOS.sdk_path}", *std_cmake_args
+      system "make"
+      bin.install "katago"
+      (etc / "katago").install "configs"
+    end
+    pkgshare.install resource("15b-network")
+    pkgshare.install resource("20b-network")
+  end
+
+  test do
+    system "#{bin}/katago", "version"
+    assert_match /All tests passed$/, shell_output("#{bin}/katago runtests").strip
```

Comment:
Is this something people will actually run as a common use case?

Reply:
People will probably not run the runtests command regularly, except when they see problems.

However, the code exercised by the tests is the basic math code that is executed all the time when actually loading and executing neural networks to play go. So I think this is a good compromise to check that the compile was correct.

It would be possible to build a true end to end test, that analyses a SGF file or tries to play a game of go via the gpt protocol, but I think that would be overkill here (for now).

I am in contact with the developers, and he suggested that the tests executed by `runtests` would be a good basis. See https://github.com/lightvector/KataGo/issues/112#issuecomment-580552392 for details.

Do you require additional tests for this to be merged?

Reply:
@SMillerDev Completely concur with @dwt here. This is a very good compromise between actually "using" the app as part of the test (which is impossible because, as I described in detail above, the CI pipeline doesn't have an OpenCL device) versus having a test that actually determines whether the build works. The `runtests` command is basically tailor-made for the purpose of ensuring a build is sane and operating within normal parameters, as you can see from @dwt's ticket with the maintainer of KataGo.